torch/distributed/tensor/_utils.py

```diff
@@ -246,6 +246,58 @@ def compute_global_tensor_info(
     return tensor_shape, tensor_stride
 
 
+def compute_global_tensor_shape(
+    shape: torch.Size, mesh: DeviceMesh, placements: Sequence[Placement]
+) -> torch.Size:
+    """
+    Compute the global size of a DTensor from the given local tensor shape,
+    the mesh and placements.
```

Comment:
let's elaborate on how this function is different from `compute_global_tensor_info`: this util collects the tensor shard shapes instead of assuming that sharding is even, and the reason we need this util is we want to have a way to calculate the full tensor shape from shards' shapes.